### PR TITLE
frameworks and credential

### PR DESCRIPTION
**Pull request by DavidHODs**

### CODE REVIEW

File: sdk.go [hunk 1]
```diff
@@ -164,3 +167,97 @@ func (c *Client) ToolkitMiddleware(next http.Handler) http.Handler {
...
+			RequestBodyStr:    string(reqBuf),
+			ResponseBodyStr:   string(resBody),
+			RequestBody:       (reqBuf),
+			ResponseBody:      (resBody),
+			StatusCode:        recRes.StatusCode,
+			Duration:          since,
+			DurationMicroSecs: since.Microseconds(),
+		}
+
+		pretty.Println(payload)
```
Comment (smilecs) [inline sdk.go:213]: Don’t forget to remove println

File: sdk.go [hunk 2]
```diff
@@ -164,3 +167,97 @@ func (c *Client) ToolkitMiddleware(next http.Handler) http.Handler {
...
+			e.Request().Body = ioutil.NopCloser(bytes.NewBuffer(reqBuf))
+
+			rec := httptest.NewRecorder()
+			start := time.Now()
+
+			recRes := rec.Result()
+			
+			for k, v := range recRes.Header {
+				for _, vv := range v {
+					e.Response().Header().Add(k, vv)
```
Comment (smilecs) [inline sdk.go:232]: Not really understanding the logic here till line 238, could you explain it to me a bit🙂

**Review by smilecs — COMMENTED** [review]
Just checkout the pr, nice stuff, just asked a little question.

**Comment by DavidHODs** [comment]
Smile

I'm so so sorry

I've not checked my mail for a while cause things had slowed down

I received an important mail tonight, scrolled down and saw review
notifications.

I didnt ignore as a disrespect...

I'll give you updates this weekend

**Pull request closed without merging**